### ISSUE TITLE
Programatically Generate List of Websites/Services to Check with Github Developer API

### ISSUE THREAD

**Issue by matthewcn56** [uclaacm/status-check#2]
Using the [Github Developer API](https://docs.github.com/en/rest/reference/repos#get-all-repository-topics), we can get a list of repos associated with a topic, and we can use the [homepage](https://api.github.com/repos/uclaacm/status-check) field, description field, and more from Github's API and more to programatically generate the list of websites to check!

**Comment by Tristan816la** [comment]
If I'm understanding this issue correctly, do we want to create a topic for the repos we'd like to monitor? On the repo documentation you sent, it doesn't seem to have an endpoint doing "Get all Repos by a Topic". But I'm looking at the [search documentation](https://docs.github.com/en/search-github/searching-on-github/searching-for-repositories#search-by-topic) and I believe it would work. Just want to check with you if this is the desired behavior:)

Also, this issue might require an API token from Github and I assume the owner's GitHub account is needed for generating that, do we have it already? Thanks!

**Comment by Tristan816la** [comment]
> If I'm understanding this issue correctly, do we want to create a topic for the repos we'd like to monitor? On the repo documentation you sent, it doesn't seem to have an endpoint doing "Get all Repos by a Topic". But I'm looking at the [search documentation](https://docs.github.com/en/search-github/searching-on-github/searching-for-repositories#search-by-topic) and I believe it would work. Just want to check with you if this is the desired behavior:)
> 
> Also, this issue might require an API token from Github and I assume the owner's GitHub account is needed for generating that, do we have it already? Thanks!

Another workaround I could think of would be getting all the repos of ACM and filter them based on the topics they have, that is much cleaner and don't require using search (but slower since we need to GET the topics for each repo).

**Comment by matthewcn56** [comment]
Yeah, a topic to search by would be really nice! We can also use different parts of repository information like website link, etc, to see what link we have to check the status of, as well as other helpful information. 

> Also, this issue might require an API token from Github and I assume the owner's GitHub account is needed for generating that, do we have it already?

If we use the octokit core API (similarly used within ACM's [opensource](https://github.com/uclaacm/opensource) repo!) we can use the API without a GitHub access token! We'll only need a github account token if we're using the GraphQL api, and we can set that through our config in the backend deployment as necessary.

**Comment by Tristan816la** [comment]
Sounds good! I also happened to look at octokit just now lol. It seems that we could use it directly on the client side [(example)](https://github.com/uclaacm/opensource/blob/574b4ac932f9d476a9bbf65b3ff4aabfe05f281d/pages/index.tsx#L130). I will experiment with that tmrw and see if it's necessary to set that up for our backend.

**Comment by reginawang3495** [comment]
Hi, not sure who the relevant one is, but is this ticket being done? Totally good if so, I just checked in with @LydiaChan528 and she hasn't started, but would you mind bopping me or other person on ticket next time? Just to avoid duplicated work :) Thanks!

**Comment by matthewcn56** [comment]
Yeah this will probably be done by the dev team and added into our backend, sorry for not making this clear! @reginawang3495

**Comment by matthewcn56** [comment]
> Sounds good! I also happened to look at octokit just now lol. It seems that we could use it directly on the client side [(example)](https://github.com/uclaacm/opensource/blob/574b4ac932f9d476a9bbf65b3ff4aabfe05f281d/pages/index.tsx#L130). I will experiment with that tmrw and see if it's necessary to set that up for our backend.

Actually, with Next.js, [`GetStaticProps`](https://nextjs.org/docs/basic-features/data-fetching/get-static-props) is ran server-side during page generation opposed to client-side. I think for separation of concerns, it might be better to keep it inside of our backend, but that's just my opinion!

**Comment by Tristan816la** [comment]
I would like to follow up on the issue. I was wondering if there are ways to get the websites we want to monitor by just using Octokit. The current situation is that I'm able to get the URLs for each repo (https://github.com/uclaacm/icpc-website) but not able to get the URLs of the official website for each committee (e.g. https://icpc.uclaacm.com/). Is there a consistent pattern or tag we could use to generate the list of websites? For instance, should we use the About section of the repo? 
![image](https://user-images.githubusercontent.com/52658897/154391388-77c0a4b1-eec5-4de3-92fd-95f26aa5160c.png)
